src/commands/registration/stor.js

```diff
@@ -37,23 +37,27 @@ module.exports = {
       });
 
       const socketPromise = new Promise((resolve, reject) => {
-        this.connector.socket.on('data', (data) => {
-          if (this.connector.socket) this.connector.socket.pause();
-          if (stream && stream.writable) {
-            stream.write(data, () => this.connector.socket && this.connector.socket.resume());
-          }
-        });
-        this.connector.socket.once('end', () => {
-          if (stream.listenerCount('close')) stream.emit('close');
-          else stream.end();
-          resolve();
-        });
+        if (pipeAllowed) this.connector.socket.pipe(stream);
+        else  this.connector.socket.on('data', (data) => {
+                if (this.connector.socket) this.connector.socket.pause();
+                if (stream && stream.writable) {
+                  stream.write(data, () => { 
+                    this.connector.socket && this.connector.socket.resume();
+                  });
+                }
+              });
+       
+              this.connector.socket.once('end', () => {
+                if (stream.listenerCount('close')) stream.emit('close');
+                else stream.end();
+                resolve();
+              });
```

Comment:
Thanks Yaser, the logic looks good to me; though I'm not sure about the `else` without the enclosing braces `{}`. 

By how it is indented, it looks like you wanted both the `socket.on` and `socket.once` to be executed on the `else` case, but only the `socket.on` will be considered because the braces are missing.